source/blog/2017-08-04-ovirt-webadmin-gwt-debug-quick-refresh.html.md

```diff
@@ -0,0 +1,132 @@
+---
+title: oVirt Webadmin GWT Debug Quick Refresh
+author: gshereme
+tags: GWT, webadmin, debugging
+date: 2017-08-04 15:00:00 CET
+comments: true
+published: true
+---
+
+As a developer, one drawback of using GWT for the oVirt Admin Portal (aka webadmin) is that the GWT compile process takes an exceptionally long time. If you make a change in some code and rebuild the ovirt-engine project using `make install-dev ...`, you'll be waiting several minutes to test your change. In practice, such a long delay in the usual code-compile-refresh-test cycle would be unbearable.
+
+READMORE
+
+Luckily, we can use GWT Super Dev Mode ("SDM") to start up a quick refresh-capable instance of the application. With SDM running, you can make a change in GWT and test the refreshed change within seconds.
+
+If you want to step through code and use the Chrome debugger, oVirt and SDM don't work well together for debugging due to Admin Portal's code and source map size. Therefore, we demonstrate how to disable source maps below.
```

Comment:
Suggest:
"Therefore, below we demonstrate how to disable source maps."
"to the oVirt Administration Portal's code"